pkg/ibmcsidriver/controller_helper.go

```diff
@@ -329,8 +334,32 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 
 	//TODO port the code from VPC BLOCK to find region if zone is given
 
-	//If the zone is not provided in storage class parameters then we pick from the Topology
-	if len(strings.TrimSpace(volume.Az)) == 0 {
+	if volume.VPCVolume.Profile != nil {
+		// validate bandwidth for dp2 profile
+		if volume.VPCVolume.Profile.Name == DP2Profile && volume.VPCVolume.Bandwidth > 0 {
```

Comment:
why we are checking Bandwidth for dp2 profile as its not required

Reply:
@arahamad If we remove this check then if some user has set bandwidth for dp2 profile to any non-zero value then we need to either log warning and reset it to zero or library will go ahead and invoke RIAAS API call. The intention is to avoid any RIAAS API call if it is going to fail